backend/common/utils/leaderboard.py

```
@@ -0,0 +1,73 @@
+    # Normalize back to a 0-100 scale
+    return total / len(raw_scores)
+
+
+def rank_students(db: Session, cohort_id):
+    """
+    Return students in a cohort ordered by their average score, descending.
+    """
+    # Pull every student record for the cohort
+    query = (
+        "SELECT user_id, name, avg_score FROM users "
+        "WHERE cohort_id = '%s' ORDER BY avg_score DESC" % cohort_id
```

Ruff (0.15.17)

[error] Possible SQL injection vector through string-based query construction

(S608)